src/components/datepicker/DatePickerInput.tsx

```diff
@@ -0,0 +1,189 @@
+import DatePicker from '.';
+import Icon from '../icon';
+import Input from '../input';
+import Popover from '../popover';
+import cx from 'classnames';
+import theme from './theme.css';
+import { formatDate } from './localeUtils';
+import { IconCalendarSmallOutline } from '@teamleader/ui-icons';
+import { DateTimeFormatOptions } from 'luxon';
+import { BoxProps } from '../box/Box';
+import { GenericComponent } from '../../@types/types';
+
+interface DatePickerInputProps extends Omit<BoxProps, 'size' | 'onChange'> {
+  /** A class name for the wrapper to give custom styles. */
+  className?: string;
+  /** Object with props for the DatePicker component. */
+  dayPickerProps?: DayPickerProps;
+  /** A footer component, rendered at the bottom of the date picker */
+  footer?: ReactNode;
+  /** A custom function to format a date. */
+  formatDate?: (selectedDate: Date, locale: string) => string;
+  /** Object with props for the Input component. */
+  inputProps?: TSFixMe; // TODO: change this to our input props
+  /** If true, component will be rendered in inverse mode. */
+  inverse?: boolean;
+  /** The language ISO locale code ('en-GB', 'nl-BE', 'fr-FR',...). */
+  locale?: string;
+  /** Callback function that is fired when the date has changed. */
+  onChange?: (selectedDate: Date) => void;
+  /** Callback function that is fired when the popover with the calendar gets closed (unfocused) */
+  onBlur?: () => void;
+  /** Object with props for the Popover component. */
+  popoverProps?: TSFixMe; // TODO: change this to our popover props
+  /** The current selected date. */
+  selectedDate?: Date;
+  /** Size of the Input & DatePicker components. */
+  size?: 'small' | 'medium' | 'large';
+  /** Overridable size of the Input component. */
+  inputSize?: 'small' | 'medium' | 'large';
+  /** Overridable size of the DatePicker component. */
+  datePickerSize?: 'small' | 'medium' | 'large';
+  /** Whether the picker should automatically open on input focus. True by default. */
+  openPickerOnFocus?: boolean;
+}
+
+interface DayPickerProps {
+  numberOfMonths: number;
+  showOutsideDays: boolean;
+  showWeekNumbers: boolean;
+  withMonthPicker: boolean;
+}
+
+const DatePickerInput: GenericComponent<DatePickerInputProps> = ({
+  className,
+  dayPickerProps,
+  footer,
+  inverse = false,
+  inputProps,
+  locale = 'en-GB',
+  popoverProps,
+  size = 'medium',
+  inputSize,
+  datePickerSize,
+  formatDate: customFormatDate,
+  openPickerOnFocus = true,
+  onChange,
+  onBlur,
+  ...others
+}) => {
+  const [isPopoverActive, setIsPopoverActive] = useState(false);
+  const [popoverAnchorEl, setPopoverAnchorEl] = useState<EventTarget | null>(null);
+  const [selectedDate, setSelectedDate] = useState<Date | undefined>(others.selectedDate);
+
+  const getFormattedDate = () => {
+    if (!selectedDate) {
+      return '';
+    }
+
+    if (!customFormatDate) {
+      return formatDate(selectedDate, locale as DateTimeFormatOptions);
```

Comment:
Q: Is `as DateTimeFormatOptions` needed here ?

Reply:
Not anymore apparently